apps/dashboard/providers/query-time-provider.tsx

```diff
@@ -0,0 +1,46 @@
+"use client";
+
+import { type ReactNode, createContext, useContext, useEffect, useState } from "react";
+
+let queryTimestamp = Date.now();
+let subscribers: ((timestamp: number) => void)[] = [];
```

Comment:
subscribers are never evicted, only new ones are added
I'm not sure if that's a problem, but it could be one in theory

Reply:
nvm, I read it wrong